main/src/library/Regex.flix

```diff
@@ -121,91 +104,90 @@ mod Regex {
+    pub def compileWithFlags(regex: String, flags: Set[Flag]): Result[String, Regex] =
         Result.tryCatch(_ -> {
             import static java.util.regex.Pattern.compile(String, Int32): ##java.util.regex.Pattern \ {};
-            Pattern(compile(regex, sumFlags(flags)))
+            compile(regex, sumFlags(flags))
         })
 
 
     ///
-    /// Return the unmatchable pattern - a pattern that will not match any input.
+    /// Return the unmatchable Regex - a regular expression that will not match any input.
     ///
-    pub def unmatchable(): Pattern =
+    pub def unmatchable(): Regex =
         try {
             import static java.util.regex.Pattern.compile(String): ##java.util.regex.Pattern \ {};
-            Pattern(compile("^\\b$"))
+            compile("^\\b$")
         } catch {
             case _: ##java.lang.Exception => unreachable!()
         }
 
     ///
-    /// Return the regular expression that matches the the literal string `s`.
+    /// Return the regular expression string that matches the the literal string `s`.
     ///
     pub def quote(s: String): String =
         import static java.util.regex.Pattern.quote(String): String \ {};
         quote(s)
 
     ///
-    /// Return the regular expression used to build this pattern.
+    /// Return the regular expression string used to build this Regex.
     ///
-    pub def pattern(patt: Pattern): String =
+    pub def pattern(rx: Regex): String =
         import java.util.regex.Pattern.pattern(): String \ {};
-        let Pattern(p1) = patt;
-        pattern(p1)
+        pattern(rx)
 
     ///
-    /// Return the flags used to build this pattern.
+    /// Return the flags used to build the Regex `rx`.
     ///
-    pub def flags(patt: Pattern): Set[Flag] =
+    pub def flags(rx: Regex): Set[Flag] =
         import java.util.regex.Pattern.flags(): Int32 \ {};
-        let Pattern(p1) = patt;
-        let i = flags(p1);
+        let i = flags(rx);
         listFlags(i) |> List.toSet
 
 
     ///
-    /// Returns `true` if the entire string `s` is matched by the pattern `patt`.
+    /// Returns `true` if the entire string `s` is matched by the Regex `rx`.
     ///
-    pub def isMatch(patt: Pattern, s: String): Bool = region rc {
+    pub def isMatch(rx: Regex, s: String): Bool = region rc {
         import java.util.regex.Matcher.matches(): Bool \ rc;
-        let Matcher(m1) = newMatcher(rc, patt, s);
+        let Matcher(m1) = newMatcher(rc, rx, s);
         matches(m1)
     }
 
     ///
-    /// Returns `true` if the string `input` is matched by the regular expression pattern `p`
+    /// Returns `true` if the string `input` is matched by the Regex `rx`
     /// at any position within the string `s`.
     ///
-    pub def isSubmatch(patt: Pattern, s: String): Bool = region rc {
-        newMatcher(rc, patt, s) |> find!
+    pub def isSubmatch(rx: Regex, s: String): Bool = region rc {
```

Comment:
We also use `rx` for `Receiver` channels, but I guess its fine.

Reply:
Certainly `Regex` and `rx` is better than `Pattern` which could be confusing.

Reply:
I thought we might have `rx` elsewhere - I'll change to `rgx`.

Reply:
Changed to `rgx`